utils/src/test/java/org/apache/cloudstack/utils/volume/VirtualMachineDiskInfoTest.java

```diff
@@ -0,0 +1,46 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+package org.apache.cloudstack.utils.volume;
+
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonParseException;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class VirtualMachineDiskInfoTest {
+
+    @Test
+    public void testGetControllerFromDeviceBusName() {
+        VirtualMachineDiskInfo vmDiskInfo = new VirtualMachineDiskInfo();
+        vmDiskInfo.setDiskDeviceBusName("scsi0:0");
+        vmDiskInfo.setDiskChain(new String[]{"[somedatastore] i-3-VM-somePath/ROOT-1.vmdk"});
+        Assert.assertEquals(vmDiskInfo.getControllerFromDeviceBusName(), "scsi");
+    }
```

Comment:
Please add test cases to pass null and blank strings, as well as, a strings without a ":" into the method.